### PR TITLE
Added middleware for requests route

### PR DESCRIPTION
**Pull request by sahsisunny**
**Issue:** 
- https://github.com/Real-Dev-Squad/website-backend/issues/1846

**Description**: 
This pull request introduces essential middleware for OOO Request API. The middleware plays a crucial role in validating incoming requests, ensuring that they meet specified criteria.

**Test Coverage:**

| File Name            | Statements (%) | Branches (%) | Functions (%) | Lines (%) | Total Covered |
|----------------------|-----------------|--------------|---------------|-----------|---------------|
| requests.ts          | 93.75           | 50           | 100           | 92.85     | 19            |
| oooRequests.ts       | 100             | 100          | 100           | 100       |               |

### CODE REVIEW

File: middlewares/validators/oooRequests.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+) => {
+  const schema = joi
+    .object()
+    .strict()
+    .keys({
+      from: joi
+        .number()
+        .min(Date.now())
+        .messages({
+          "number.min": "from date must be greater than or equal to current date",
```
Comment (iamitprakash) [inline middlewares/validators/oooRequests.ts:19]: Today's date

File: middlewares/validators/requests.ts [hunk 2]
```diff
@@ -1,135 +1,28 @@
...
-      createdAt: joi.number().optional(),
-      updatedAt: joi.number().optional(),
-    });
+  const type = req.body.type;
 
   try {
-    await schema.validateAsync(req.body, { abortEarly: false });
+    switch (type) {
+      case REQUEST_TYPE.OOO:
+        await createOooStatusRequestValidator(req as OooStatusRequestRequest, res as OooStatusRequestResponse, next);
```
Comment (iamitprakash) [inline middlewares/validators/requests.ts:16]: what is this..?

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: middlewares/validators/oooRequests.ts (hunk 1 above)
Comment (sahsisunny) [reply middlewares/validators/oooRequests.ts]: Done

File: middlewares/validators/requests.ts (hunk 2 above)
Comment (sahsisunny) [reply middlewares/validators/requests.ts]: OOO request type of the request.

**Review by iamitprakash — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**